tests/MongoDB.Driver.Tests/ConnectionsSurvivePrimaryStepDownTests.cs

```diff
@@ -57,12 +59,19 @@ public class ConnectionsSurvivePrimaryStepDownTests
 
                 using (ConfigureFailPoint(client, errorCode))
                 {
-                    var exception = Record.Exception(() => { collection.InsertOne(new BsonDocument("test", 1)); });
+                    var document = new BsonDocument("test", 1);
+                    var exception = Record.Exception(() => { collection.InsertOne(document); });
 
                     var e = exception.Should().BeOfType<MongoNodeIsRecoveringException>().Subject;
                     e.Code.Should().Be(errorCode);
+
+                    eventCapturer.Next().Should().BeOfType<ConnectionPoolClearedEvent>();
+                    eventCapturer.Events.Should().BeEmpty();
+
+                    collection.InsertOne(document);
```

Comment:
It may not matter for this particular test... but in general you want to use a brand new document for each call to InsertOne.

The reason is subtle... the call to InsertOne on line 63 MODIFIED the document (even if the InsertOne then subsequently failed), so the second call to InsertOne on line 71 is no longer inserting `{ test : 1 }` but rather `{ _id : ObjectId(...), test : 1 }` which is not the intent.

Reply:
done